scala/private/rule_impls.bzl

```diff
@@ -1014,68 +1014,25 @@ def _pack_source_jars(ctx):
     #_pack_source_jar may return None if java_common.pack_sources returned None (and it can)
     return [source_jar] if source_jar else []
 
-def scala_binary_impl(ctx):
-    scalac_provider = _scalac_provider(ctx)
-    unused_dependency_checker_mode = get_unused_dependency_checker_mode(ctx)
-    unused_dependency_checker_is_off = unused_dependency_checker_mode == "off"
-
-    jars = _collect_jars_from_common_ctx(
-        ctx,
-        scalac_provider.default_classpath,
-        unused_dependency_checker_is_off = unused_dependency_checker_is_off,
-    )
-    (cjars, transitive_rjars) = (jars.compile_jars, jars.transitive_runtime_jars)
-
-    wrapper = _write_java_wrapper(ctx, "", "")
-
-    executable = _declare_executable(ctx)
-
-    out = _scala_binary_common(
-        ctx,
-        executable,
-        cjars,
-        transitive_rjars,
-        jars.transitive_compile_jars,
-        jars.jars2labels,
-        wrapper,
-        unused_dependency_checker_ignored_targets = [
-            target.label
-            for target in scalac_provider.default_classpath +
-                          ctx.attr.unused_dependency_checker_ignored_targets
-        ],
-        unused_dependency_checker_mode = unused_dependency_checker_mode,
-        deps_providers = jars.deps_providers,
-    )
-    _write_executable(
-        ctx = ctx,
-        executable = executable,
-        jvm_flags = ctx.attr.jvm_flags,
-        main_class = ctx.attr.main_class,
-        rjars = out.transitive_rjars,
-        use_jacoco = False,
-        wrapper = wrapper,
-    )
-    return out
-
 def scala_repl_impl(ctx):
-    scalac_provider = _scalac_provider(ctx)
+    _scalac_provider = scalac_provider(ctx)
```

Comment:
why did you add the `_` on the local var? you removed it in other cases? can we just use a different name?

maybe the function should be `get_scalac_provider(ctx)` if it is going to be public?

Reply:
It looks like he stuck with file reorganization while keeping any function/variable renames to a minimum. A lot of these now public methods should be refactored further-- tackling the name right now won't matter too much.

Reply:
Agree with you @johnynek - I think `get_scalac_provider` is probably the right move here, I just changed things to get it to build properly. I'll follow up with that instead.